flake.nix

```diff
@@ -0,0 +1,35 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-parts = {
+      url = "github:hercules-ci/flake-parts";
+      inputs.nixpkgs-lib.follows = "nixpkgs";
+    };
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+  outputs = inputs @ {
+    nixpkgs,
+    flake-parts,
+    rust-overlay,
+    ...
+  }:
+    flake-parts.lib.mkFlake {inherit inputs;} {
+      systems = nixpkgs.lib.systems.flakeExposed;
+      perSystem = {system, ...}: let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [rust-overlay.overlays.default];
+        };
+        rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      in {
+        formatter = pkgs.alejandra;
+        devShells.default = pkgs.mkShell {
+          nativeBuildInputs = [rust pkgs.pkg-config];
```

Comment:
nativeBuildInputs/buildInputs separation makes little sense for devShell, use packages argument instead.

Reply:
Done